### PR TITLE
Expand home within the path in couple of places moar

### PR DESCRIPTION
**Pull request by Sija**
Some places seem unjustly forgotten, followup to #7903.

### CODE REVIEW

**Comment by straight-shoota** [comment]
Why do we need home expansion in these places? There should be a clear reason for each of them. If we would just apply `home: true` randomly across stlib and compiler sources, we could've left it as default as well.

See also discussion in #8756. This is similar.

File: src/http/server/handlers/static_file_handler.cr [hunk 1]
```diff
@@ -19,7 +19,7 @@ class HTTP::StaticFileHandler
   # If *directory_listing* is `false`, directory listing is disabled. This means that
   # paths matching directories are ignored and next handler is called.
   def initialize(public_dir : String, fallthrough = true, directory_listing = true)
-    @public_dir = Path.new(public_dir).expand
+    @public_dir = Path.new(public_dir).expand(home: true)
```
Comment (Sija) [inline src/http/server/handlers/static_file_handler.cr]: I'd expect that passing a `public_dir` will be expanded, incl. home.

File: src/compiler/crystal/compiler.cr [hunk 2]
```diff
@@ -381,13 +381,13 @@ module Crystal
         end
       end
 
+      output_filename = File.expand_path(output_filename, home: true)
```
Comment (Sija) [inline src/compiler/crystal/compiler.cr]: I'd expect that calling `-o ~/foo.obj` would work.

File: src/compiler/crystal/command.cr [hunk 3]
```diff
@@ -501,10 +501,10 @@ class Crystal::Command
 
   private def gather_sources(filenames)
     filenames.map do |filename|
+      filename = File.expand_path(filename, home: true)
```
Comment (Sija) [inline src/compiler/crystal/command.cr]: ditto, I'd expect expanding filepaths passed to the _compile_ command.

File: src/compiler/crystal/compiler.cr (hunk 2 above)
Comment (asterite) [reply src/compiler/crystal/compiler.cr]: And it works: the shell expands it for you.

Comment (Sija) [reply src/compiler/crystal/compiler.cr]: yeah, unless it doesn't ;)

**Comment by straight-shoota** [comment]
Expanding CLI arguments is a task for the shell. If your shell doesn't support that, you can use some other means (for example `$HOME`). That's not a feature that needs to be supplemented by the compiler.

**Comment by straight-shoota** [comment]
Expanding home is intended for paths that are *not* passed from the shell, for example in configuration files, external requests or any kind of direct user input.

**Comment by Sija** [comment]
@straight-shoota I agree, thus I've removed my changes except the one re: static file handler.

**Comment by straight-shoota** [comment]
I don't think StaticFileHandler qualifies either.

**Comment by asterite** [comment]
I'd say, if nobody creates a bug report saying "I was expecting this to expand home" based on a real use case, we shouldn't be changing things.

**Comment by straight-shoota** [comment]
Also, it's always really easy to *enable* home expansion outside of stdlib code when you need it. The other way around doesn't work.

**Comment by Sija** [comment]
Fair 'nuff, closing.

**Pull request closed without merging**